google/cloud/aiplatform/training_jobs.py

```diff
@@ -1701,7 +1701,7 @@ def __init__(
     # TODO(b/172368070) add timestamp split, training_pipeline.TimestampSplit
     def run(
         self,
-        dataset: Optional[datasets.Dataset] = None,
+        dataset: Optional[datasets.SupportedDatasets] = None,
```

Comment:
I like using the TypeVar for our internal implementations but using the private `_Dataset` class is also sufficient for that case. For public APIs we may need a different approach. If this doesn't expand in the docs it's probably more informative to put the explicit unions `Union[datasets.TabularDataset, datasets.ImageDataset, datasets.VideoDataset, datasets.TextDataset]`. This would just be for custom training jobs.

Reply:
Removed `SupportedDatasets` and defaulted to `_Dataset` in private methods and the explicit unions in the three custom training job classes.